sdks/unreal/Agones/Source/Agones/Private/AgonesComponent.cpp

```diff
@@ -146,6 +192,45 @@ void UAgonesComponent::GameServer(const FGameServerDelegate SuccessDelegate, con
 	Request->ProcessRequest();
 }
 
+bool UAgonesComponent::DeserializeAndBroadcastWatch(FString const& JsonString)
+{
+    TSharedPtr<FJsonObject> JsonObject;
+    const TSharedPtr<FJsonObject>* ResultObject = nullptr;
+    TSharedRef<TJsonReader<TCHAR>> const JsonReader = TJsonReaderFactory<TCHAR>::Create(JsonString);
+    if (FJsonSerializer::Deserialize(JsonReader, JsonObject) && JsonObject.IsValid() &&
+        JsonObject->TryGetObjectField(TEXT("result"), ResultObject) && ResultObject->IsValid())
+    {
+        WatchGameServerDelegate.Broadcast(FGameServerResponse(*ResultObject));
+        return true;
+    }
+    return false;
+}
+
+void UAgonesComponent::HandleWatchMessage(const void* Data, SIZE_T Size, SIZE_T BytesRemaining)
+{
+    if (BytesRemaining > 0)
+    {
+        WatchMessageBuffer.Append(UTF8_TO_TCHAR(static_cast<const UTF8CHAR*>(Data)), Size);
+        return;
+    }
+
+    FString const Message = FString(Size, UTF8_TO_TCHAR(static_cast<const UTF8CHAR*>(Data)));
+
+    if (WatchMessageBuffer.Len() != 0)
+    {
+        FString CombinedBuffer;
+        CombinedBuffer.Reserve(WatchMessageBuffer.Len() + Message.Len());
+        CombinedBuffer.Append(WatchMessageBuffer);
+        CombinedBuffer.Append(Message);
+        DeserializeAndBroadcastWatch(CombinedBuffer);
+        WatchMessageBuffer.Empty();
+    }
+    else
+    {
+        DeserializeAndBroadcastWatch(Message);
+    }

```

Comment:
```suggestion
        return
    }
    
DeserializeAndBroadcastWatch(Message);
```

I will oneday rid the world of all `else` statments 👯